homework_3/calculator/internal/engine/handler/handler.go

```diff
@@ -0,0 +1,113 @@
+package handler
+
+import (
+	"context"
+	"log"
+
+	domainmodels "github.com/superbet-group/code-cadets-2021/lecture_3/03_project/calculator/internal/domain/models"
+	rabbitmqmodels "github.com/superbet-group/code-cadets-2021/lecture_3/03_project/calculator/internal/infrastructure/rabbitmq/models"
+)
+
+// Handler handles bets received and bets calculated.
+type Handler struct {
+	betRepository BetRepository
+}
+
+// New creates and returns a new Handler.
+func New(betRepository BetRepository) *Handler {
+	return &Handler{
+		betRepository: betRepository,
+	}
+}
+
+// HandleBetsReceived handles bets received.
+func (h *Handler) HandleEventsReceived(
+	ctx context.Context,
+	eventsReceived <-chan rabbitmqmodels.EventReceived,
+) <-chan rabbitmqmodels.BetCalculated{
+	resultingBets := make(chan rabbitmqmodels.BetCalculated)
+
+	go func() {
+		defer close(resultingBets)
+
+		for eventReceived := range eventsReceived {
+			log.Println("Processing event received, eventId:", eventReceived.Id)
+
+			//get bet from calc_bets db
+			domainBets, exists, err := h.betRepository.GetBetBySelectionID(ctx, eventReceived.Id)
+			if err != nil {
+				log.Println("Failed to fetch a bet which should be updated, error: ", err)
+				continue
+			}
+			if !exists {
+				log.Println("A bet which should be updated does not exist, betId: ", eventReceived.Id)
+				continue
+			}
+
+			for _, bet := range domainBets {
+
+				var payout float64
+				if eventReceived.Outcome == "won" {
```

Comment:
Try to avoid using magic strings/numbers.
I would suggest that `"won"` is a constant